front/src/applications/operationalStudies/components/MacroEditor/export.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  osrdEditoastApi,
+  type Distribution,
+  type TrainScheduleBase,
+} from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+import type { NGEDeleteEvent, NGEEvent, NGEUpdateEvent, TrainrunSectionOperation } from './types';
+
+const defautlPayloads = {
+  constraint_distribution: 'STANDARD' as Distribution,
+  rolling_stock_name: '',
+  start_time: '2024-07-15T08:00:00+02:00',
+};
+
+const ngeCreateOperation = (trainrunSection: TrainrunSectionOperation) => {
+  const { sourceNode, targetNode } = trainrunSection;
+  const [sourceTrigram, sourceSecondaryCode] = sourceNode.betriebspunktName.split('/');
+  const [targetTrigram, targetSecondaryCode] = targetNode.betriebspunktName.split('/');
+  const payloads: TrainScheduleBase[] = [
+    {
+      ...defautlPayloads,
+      path: [
+        {
+          trigram: sourceTrigram,
+          secondary_code: sourceSecondaryCode,
+          id: `${sourceNode.id}`,
+          deleted: false,
+        },
+        {
+          trigram: targetTrigram,
+          secondary_code: targetSecondaryCode,
+          id: `${targetNode.id}`,
+          deleted: false,
+        },
+      ],
+      train_name: trainrunSection.trainrun.name,
+    },
+  ];
+  return payloads;
+};
+
+const ngeUpdateOperation = (trainrunSections: TrainrunSectionOperation[]) => {
+  const path = trainrunSections.flatMap((section, index) => {
```

Comment:
Since non-continous trainruns (I believe you had that in mind) exist in NGE, we need to prevent that. However it is true that it is not our priority (but soon to be)

Reply:
[A B] [C D] is a case of discontinuous trainruns (a single train run made up of two separate paths).

[B C] [A B] is a case of a continuous train run, but with the sections in an arbitrary order.